File: 251004/Krotsyuk/module-discussion/src/main/java/by/ikrotsyuk/bsuir/modulediscussion/mapper/ReactionMapper.java
package by.ikrotsyuk.bsuir.modulediscussion.mapper;

import by.ikrotsyuk.bsuir.kafkacommunicationparts.events.dto.KafkaReactionResponseEventDTO;
import by.ikrotsyuk.bsuir.kafkacommunicationparts.events.dto.ReactionDTO;
import by.ikrotsyuk.bsuir.modulediscussion.dto.request.ReactionRequestDTO;
import by.ikrotsyuk.bsuir.modulediscussion.dto.response.ReactionResponseDTO;
import by.ikrotsyuk.bsuir.modulediscussion.entity.ReactionEntity;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingConstants;

import java.util.List;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
public interface ReactionMapper {
    ReactionResponseDTO toDTO(ReactionEntity reactionEntity);

    ReactionEntity toEntity(ReactionRequestDTO reactionRequestDTO);

    List<ReactionResponseDTO> toDTOList(List<ReactionEntity> reactionEntityList);

    @Mapping(target = "articleId", source = "articleId")
    ReactionEntity toEntity(ReactionDTO reactionDTO, Long articleId);

    @Mapping(target = "reactionId", source = "id")
    KafkaReactionResponseEventDTO toReactionResponse(ReactionEntity reactionEntity);

    List<KafkaReactionResponseEventDTO> toReactionResponseList(List<ReactionEntity> reactionEntityList);

    ReactionEntity toKafkaEntity(ReactionDTO reactionDTO);
}

File: 251001/Vasilenko/discussion/src/main/kotlin/org/ex/discussion/mapper/ReactionMapper.kt
package org.ex.discussion.mapper

import org.ex.discussion.dto.request.ReactionRequestDTO
import org.ex.discussion.dto.response.ReactionResponseDTO
import org.ex.discussion.model.Reaction
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants.ComponentModel

@Mapper(
    componentModel = ComponentModel.SPRING,
    injectionStrategy = InjectionStrategy.CONSTRUCTOR
)
interface ReactionMapper {

    fun toDto(entity: Reaction): ReactionResponseDTO

    fun toEntity(dto: ReactionRequestDTO): Reaction
}